proposals/NNNN-swiftpm-package-resolved-lock-file.md

```diff
@@ -0,0 +1,238 @@
+# Treating `Package.resolved` as a lock file
+
+* Proposal: [SE-NNNN](https://github.com/apple/swift-evolution/pull/2212/files?short_path=14ad3ef#diff-14ad3ef424e6ca9cf3077fd1a3c0e898a34a2b14523c1358edbbf6fa1a5def16)
+* Authors: [Franz Busch](https://github.com/FranzBusch)
+* Status: **Pitch**
+* Implementation: Awaiting implementation
+
+## Introduction
+
+The Swift package manager is an essential part of the ecosystem and provides
+functionality to download your dependencies, version them and build your
+packages. Furthermore, the Swift package manager provides a locking mechanism
+for dependencies to avoid accidental updates. Locking of dependencies has become
+more relevant in the past years due to many different supply chain attacks in
+multiple language ecosystems. 
+
+## Motivation
+
+Since, the early days of the Swift package manager it had a couple of essential
+commands to manage the dependencies of a package and building the package. These
+commands are `swift package resolve`, `swift package update` and `swift build`. Over
+the years, those commands gained more and more options to customize their
+behaviour; however, there doesn't exist a clear definition when each of those
+commands resolves or updates dependencies and how this interacts with the
+presence or absence of a `Package.resolved` file. Furthermore, a lot of the
+commands don't treat the `Package.resolved` file as a lock file and often update
+its contents.
+
+This proposal aims to define a clear behavior for the Swift package manager when
+it resolves and updates dependencies.
+
+## Proposed solution
+
+### Deprecated: swift package resolve
+
+We propose to deprecate this command since it is confusing in behavior. It
+currently does both a resolution and download. Additionally, other commands may
+trigger a resolution as well. We propose to replace it with swift package fetch
+below. 
+
+### swift package fetch
+
+Fetches the dependencies of a package.
+
+#### Description
+
+If a `Package.resolved` file is present, this command will fetch all dependencies
```

Comment:
Nit: Restating `swift package fetch` instead of `this command` would make this sentence easier to read.